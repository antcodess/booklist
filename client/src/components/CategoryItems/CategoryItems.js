import React from 'react';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

import './CategoryItems.scss'; 

export const CategoryItems = ({ categoryItems, categoryTitle }) => {

  return (
    <React.Fragment>
      <BreadCrumbNav title={categoryTitle} />
      <section className='category-items-container'>
        <h2>{ categoryTitle }</h2>
        <div className='category-items'>
          {categoryItems ? 
            categoryItems.map(item => <DirectoryItem key={item.id} item={item} />)
            : null }
        </div>
      </section>
    </React.Fragment> 
  )
};

export default CategoryItems;