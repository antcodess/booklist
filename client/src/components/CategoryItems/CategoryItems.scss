.category-items-container {
  margin-top: 20px;
  
  h2 {
    text-align: left;
    margin-bottom: 40px; 
    font-size: 2rem;
  }
  
  .category-items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 10px;
    
    .directory-item {
      margin-bottom: 20px;
    }
  }
} 