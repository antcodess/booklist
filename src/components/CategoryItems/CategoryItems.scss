.category-items-container {
  margin-top: 20px;
  
  h2 {
    text-align: left;
    margin-bottom: 40px; 
  }
  .category-items {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr 1fr;
    grid-gap: 10px;
    
    .directory-item {
      margin-bottom: 20px;
    }
  }
} 