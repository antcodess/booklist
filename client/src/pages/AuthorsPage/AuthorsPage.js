import React from 'react';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';

import CategoryItems from '../../components/CategoryItems/CategoryItems';

import { titleCase } from '../../components/CategoryItems/CategoryItems.utils';
import { selectAuthorItems } from '../../redux/shop/shop.selectors';

const AuthorsPage = ({ match, authorItems }) => {
  const authorName = titleCase(decodeURIComponent(match.params.authorName).replace(/[-]/g, ' '));
  
  return (
    <Route 
      path={`${match.path}`} 
      render={(props) => <CategoryItems {...props} categoryItems={authorItems} categoryTitle={authorName} /> } />
  )
};

const mapStateToProps = (state, ownProps) => ({
  authorItems: selectAuthorItems(decodeURIComponent(ownProps.match.params.authorName).replace(/[-]/g, ' '))(state)
});

export default connect(mapStateToProps)(AuthorsPage);