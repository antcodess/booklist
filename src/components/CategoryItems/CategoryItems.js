import React from 'react';
import { connect } from 'react-redux';

import BreadCrumbNav from '../BreadCrumbNav/BreadCrumbNav';
import DirectoryItem from '../DirectoryItem/DirectoryItem';

import { titleCase } from './CategoryItems.utils';
import { selectCategoryItems } from '../../redux/shop/shop.selectors';

import './CategoryItems.scss'; 

const CategoryItems = ({ match, categoryItemsList }) => {
  const categoryItems = categoryItemsList[0];
  const categoryTitle = titleCase(decodeURIComponent(match.params.categoryTitle).replace(/[-]/g, ' '));

  return (
    <React.Fragment>
      <BreadCrumbNav title={categoryTitle} />
      <section className='category-items-container'>
        <h2>{ categoryTitle }</h2>
        <div className='category-items'>
          {categoryItems ? 
            categoryItems.map(item => <DirectoryItem key={item.id} item={item} />)
            : null }
        </div>
      </section>
    </React.Fragment>
  )
};

const mapStateToProps = (state, ownProps) => ({
  categoryItemsList: selectCategoryItems(decodeURIComponent(ownProps.match.params.categoryTitle).replace(/[-]/g, ' '))(state)
});

export default connect(mapStateToProps)(CategoryItems);