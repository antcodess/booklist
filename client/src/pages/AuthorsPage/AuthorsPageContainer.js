import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';

import { selectIsProductFetching } from '../../redux/shop/shop.selectors';

import WithSpinner from '../../components/WithSpinner/WithSpinner';
import AuthorsPage from './AuthorsPage';

const mapStateToProps = createStructuredSelector({
  isLoading: selectIsProductFetching
});

const AuthorsPageContainer = compose(
  connect(mapStateToProps),
  WithSpinner
)(AuthorsPage);

export default AuthorsPageContainer;  