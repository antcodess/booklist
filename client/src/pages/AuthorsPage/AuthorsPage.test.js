import React from 'react';
import { shallow } from 'enzyme';
import { AuthorsPage } from './AuthorsPage';
import CategoryItems from '../../components/CategoryItems/CategoryItems';

describe('AuthorsPage Component', () => {
  let wrapper;
  let mockMatch;
  let mockAuthorItems;
  
  beforeEach(() => {

    mockMatch = {
      params: {
        authorName: 'Sample Author'
      }
    };

    mockAuthorItems = [{ id: 1, authorName: 'Sample Author'}];
    const mockProps = {
      match: mockMatch, 
      authorItems: mockAuthorItems
    };

    wrapper = shallow(<AuthorsPage { ...mockProps } />);
  });

  it('should render AuthorsPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});