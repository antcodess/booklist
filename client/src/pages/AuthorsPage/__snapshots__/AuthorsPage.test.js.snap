// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorsPage Component should render AuthorsPage Component 1`] = `ShallowWrapper {}`;
